amazonia/schema.yaml

```diff
@@ -0,0 +1,309 @@
+        empty: False
+        regex: '^([a-zA-Z0-9]+$)' # matches an alphanumeric title
+      cidr:
+        type: 'string'
+        empty: False # Below regex matches a CIDR pattern eg: 10.0.0.0/16
+        regex: '^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/([0-9]|[1-2][0-9]|3[0-2]))$'
+protocols: # a list of strings specifying the protocols to allow access through load balancers into your autoscaling groups.
+  type: 'list'
+  empty: False
+  schema:
+    type: 'string'
+    empty: False
+instanceports: # a list of strings specifying the ports to send traffic from your load balancers into your autoscaling groups.
+  type: 'list'
+  empty: False
+  schema:
+    type: 'string'
+    empty: False
+loadbalancerports: # a list of strings specifying the ports to receive traffic on your load balancers.
+  type: 'list'
+  empty: False
+  schema:
+    type: 'string'
+    empty: False
+path2ping: # A string containing the path for your load balancers to check to confirm that your instances are healthy.
+  type: 'string'
+  empty: False
+minsize: # a string containing the minimum amount of instances for your autoscaling groups to run
+  type: 'string'
+  empty: False
+maxsize: # a string containing the maximum amount of instances for your autoscaling groups to run
+  type: 'string'
+  empty: False
+health_check_grace_period: # a string containing the amount of seconds for your load balancer to wait for new instances before checking if they are 'healthy' or not.
+  type: 'string'
+  empty: False
+health_check_type: # A string containing the type of health check to use for your autoscaling groups. This should 90% of the time be 'ELB'
+  type: 'string'
+  empty: False
+dependencies: # A list of strings with the titles of objects that your autoscaling units depend on. eg: your autoscaling group may have userdata that needs internet access, therefore they have a dependency on your NAT.
+  type: 'list'
+  nullable: True
+  schema:
+    type: 'string'
+autoscaling_units:  # A list of autoscaling groups to create. Load balancers, security groups, launch configurations, and autoscaling groups will be created for each of these units
+  type: 'list'
+  nullable: True
+  schema:
+    type: 'dict'
+    schema:
+      unit_title:  # A string containing the title of this autoscaling unit.
+        type: 'string'
+        regex: '^([a-zA-Z0-9]+$)' # matches an alphanumeric title
+        nullable: True  # For all values in an autoscaling unit. If they are null, a matching value must be
+      hosted_zone_name: # provided at the lowest level of the yaml structure (if autoscaling units are not also empty).
+        type: 'string'
+        nullable: True
+      image_id: # A string specifying which ami to use for this specific autoscaling group by its id
+        type: 'string'
+        nullable: True
+        regex: '^([a][m][i][\-][a-zA-Z0-9]+$)' # matches an ami number eg: ami-some1234
```

Comment:
Should we use this regex? ^(ami-[a-zA-Z0-9]+$) for ami ids. Its the same but a bit cleaner